### ISSUE TITLE
"checked" state doesn't change when "checked" attribute removed

### ISSUE THREAD

**Issue by viveleroi** [Matt-Esch/virtual-dom#335]
I'm generating a checkbox input with an attribute of `checked`. It shows properly in the browser as checked.

However, when the next render/diff/patch chain runs, the DOM is replaced with:

    <input type="checkbox"></input>

Yet the form element remains checked, in both Chrome and Firefox. I have a feeling this is a browser issue and not something broken in vdom, but it's a scenario I need to find a solution for.

**Comment by marcelklehr** [comment]
I believe after the first render, it doesn't matter whether you set the `checked` attribute, because vdom will either add or remove it on the input element, but won't touch its state.

Right now I'm trying to change its state by modifying the checked property, which doesn't work, however...

**Comment by viveleroi** [comment]
I discovered that vdom is hard-coded to only remove attributes when they're `undefined`. That knowledge solved the problem, but isn't as usable as I'd like.

    attributes: {
         type: 'checkbox',
         checked: node.selected() ? 'checked' : undefined
    }

https://github.com/helion3/inspire-tree/blob/master/src/dom.js#L62

**Comment by mtyaka** [comment]
Using a property and setting it to a boolean value (instead of using the `checked` attribute) should work:

    var checked = true; // or false
    h('input', {type: 'checkbox', checked: checked});

See example: https://jsfiddle.net/y0w1pzgp/

**Comment by marcelklehr** [comment]
Mh, works indeed. I'll have to dig some more to identify the problem... :)

**Comment by marcelklehr** [comment]
Yup, now it works! The problem was the value I supplied for the checked
property
